File: Python/Euler024.py
#!/usr/bin/env python3
from __future__ import print_function, division
from itertools import permutations, islice
import itertools
import sys
if sys.version.startswith('2'):
    range = xrange


def main():
    target = 1000000
    digits = list(range(0, 10))
    perms = permutations(digits)
    result = next(islice(perms, target - 1, target))
    result = ''.join(str(d) for d in result)
    print(result)


if __name__ == '__main__':
    main()

File: Ruby/Euler024.rb
#!/usr/bin/env ruby

def main
  target = 1000000
  digits = (0..9).to_a
  perms = digits.permutation.lazy
  result = perms.drop(target - 1).first
  puts result.join("")
end

if __FILE__ == $0
  main
end
